eeps/eep-00XX.md

```diff
@@ -0,0 +1,317 @@
+    Author: José Valim <jose(dot)valim(at)dashbit(dot)co>
+    Status: Active
+    Type: Process
+    Created: 09-Dec-2025
+    Post-History:
+****
+EEP XX: Partially applied functions
+----
+
+Abstract
+========
+
+This EEP proposes an alternative to "MFArgs" (Module-Function-Arguments):
+three-element tuples where the first element is a module, the second is
+a function name, and the third is a list of arguments. The proposed
+alternative preserves the desired properties of MFArgs while being more
+ergonomic and with none of MFArgs limitations.
+
+Rationale
+=========
+
+Today, the use of MFArgs are pervasive in Erlang. Generally speaking,
+an API accepts MFArgs either as tuples or as three distinct arguments.
+MFArgs can be invoked as-is but quite often they have additional arguments
+prepended, as shown below:
+
+```erlang
+{Mod, Fun, Args} = MFArgs
+apply(Mod, Fun, [SomeValue | Args]).
+```
+
+One of the main reasons MFArgs exist is because anonymous functions
+which close over an existing environment cannot be serialized across
+nodes nor be persisted to disk, so when dealing with distribution,
+disk persistence, or hot code upgrades, you must carefully stick with
+MFArgs. Similarly, configuration files do not support anonymous
+functions, and MFArgs are the main option.
```

Comment:
Passing an anonymous function with an environment over the distribution works. I must be misunderstanding something?

Reply:
Poor phrasing on my end, I should clarify it only works if you have the exact same module version (which, unless you declare a version, will change even with minor changes to the source). I will improve that on a future pass to the EEP.